src/circuit/integer/assert_not_zero.rs

```diff
@@ -0,0 +1,91 @@
+use super::IntegerChip;
+use crate::circuit::main_gate::{CombinationOption, MainGateInstructions, Term};
+use crate::circuit::{AssignedInteger, AssignedValue};
+use crate::BIT_LEN_LIMB;
+use halo2::arithmetic::FieldExt;
+use halo2::circuit::Region;
+use halo2::plonk::Error;
+use num_bigint::BigUint as big_uint;
+use std::convert::TryInto;
+
+impl<W: FieldExt, N: FieldExt> IntegerChip<W, N> {
+    pub(crate) fn _assert_not_zero(&self,
+        region: &mut Region<'_, N>,
+        a: &AssignedInteger<N>,
+        offset: &mut usize
+    ) -> Result<(), Error> {
+        let main_gate = self.main_gate();
+        let (zero, one) = (N::zero(), N::one());
+
+        // reduce result (r) should be less than 2 * wrong_modulus,
```

Comment:
Seems like it should be less than `1 << wrong_modulus_bit_lenght`

Reply:
Done